tests/tt_metal/tt_metal/perf_microbenchmark/common/util.hpp

```diff
@@ -22,12 +23,16 @@ inline uint64_t get_t0_to_any_riscfw_end_cycle(tt::tt_metal::Device *device, con
     auto device_id = device->id();
     uint64_t min_cycle = -1;
     uint64_t max_cycle = 0;
+    uint64_t dprint_base = hal.get_dev_addr(HalDispatchCoreType::HAL_CORE_TYPE_TENSIX,
+        HalMemAddrType::HAL_L1_MEM_ADDR_DPRINT);
+    dprint_buf_msg_t *dprint_msg = reinterpret_cast<dprint_buf_msg_t *>(dprint_base);
+    // This works for tensix only, will need to be updated for eth
     vector<uint64_t> print_buffer_addrs = {
-        static_cast<uint64_t>(GET_MAILBOX_ADDRESS_HOST(dprint_buf.data[DPRINT_RISCV_INDEX_NC])),
-        static_cast<uint64_t>(GET_MAILBOX_ADDRESS_HOST(dprint_buf.data[DPRINT_RISCV_INDEX_BR])),
-        static_cast<uint64_t>(GET_MAILBOX_ADDRESS_HOST(dprint_buf.data[DPRINT_RISCV_INDEX_TR0])),
-        static_cast<uint64_t>(GET_MAILBOX_ADDRESS_HOST(dprint_buf.data[DPRINT_RISCV_INDEX_TR1])),
-        static_cast<uint64_t>(GET_MAILBOX_ADDRESS_HOST(dprint_buf.data[DPRINT_RISCV_INDEX_TR2]))
+        reinterpret_cast<uint64_t>(dprint_msg->data[DPRINT_RISCV_INDEX_NC]),
+        reinterpret_cast<uint64_t>(dprint_msg->data[DPRINT_RISCV_INDEX_BR]),
+        reinterpret_cast<uint64_t>(dprint_msg->data[DPRINT_RISCV_INDEX_TR0]),
+        reinterpret_cast<uint64_t>(dprint_msg->data[DPRINT_RISCV_INDEX_TR1]),
+        reinterpret_cast<uint64_t>(dprint_msg->data[DPRINT_RISCV_INDEX_TR2]),
```

Comment:
I'm not sure how this is meant to work: `dprint_msg->data` is a 2D array of `uint8_t` and we're reinterpet_casting ipecific indices to `uint64_t`. Seems like we have 8 `uint8_t`s per core. Why not just keep a single 64 bit value per core in the array?

Reply:
I don't follow.  The buffers are larger than 8 bytes, something like 200 bytes.  This code is definitely ugly due to use using uint64_t for addresses.  This value after the case is the address of the base of the 200 bytes for that core.  I agree, it is a mess but not a new mess.  Make sense?

Reply:
Oh .. I understand now, I misread `uint64_t` as `uint64_t *`.